src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -364,6 +418,8 @@ private void RemoveAllCachedDiagnostics()
                 this.AssertIsForeground();
                 Debug.Assert(!_disposed);
 
+                _shutdownCancellationSource.Token.ThrowIfCancellationRequested();
```

Comment:
same concern.  should our UI thread checks actually just be returning instead of throwing.

Reply:
same here. sure, we can change it to return.